File: flutter_theoplayer_sdk_android/android/src/main/kotlin/com/theoplayer/flutter/transformers/TimeRangeTransformer.kt
package com.theoplayer.flutter.transformers

import com.theoplayer.android.api.timerange.TimeRanges

typealias FlutterTimeRange = com.theoplayer.flutter.pigeon.TimeRange

object TimeRangeTransformer {

    fun toFlutterTimeRanges(timeRanges: TimeRanges): List<FlutterTimeRange> {
        return timeRanges.map {
            FlutterTimeRange(it.start, it.end)
        }
    }

}

File: flutter_theoplayer_sdk/flutter_theoplayer_sdk_ios/ios/Classes/transformers/TimeRangeTransformer.swift
//
//  TimeRangeTransformer.swift
//  flutter_theoplayer_sdk_ios
//
//  Created by Hovig on 17/10/2023.
//

import Foundation
import THEOplayerSDK

struct TimeRangeTransformer {
    
    static func toFlutterTimeRanges(timeRanges:[THEOplayerSDK.TimeRange]) -> [TimeRange] {
        timeRanges.map { timeRange in
            TimeRange(start: timeRange.start, end: timeRange.end)
        }
    }
}
